admin APIs

prasidha1 opened the pull request:
Expose essential admin APIs for managing games, rate limits, and viewing player stats

jishnujp commented on backend/app/routes/admin.py:
```diff
@@ -1,17 +1,97 @@
+        games = Game.query.all()
+        return {"games": [game.name for game in games]}, 200
+
+    @admin_required
+    def post(self):
+        """Add a new game"""
+        data = request.get_json()
+        name = data.get("name")
+        description = data.get("description")
+        game_type = data.get("type")
+
+        if not name or not game_type:
+            return {"error": "Name and type are required"}, 400
+
+        if Game.query.filter_by(name=name).first():
+            return {"error": "Game with this name already exists"}, 400
+
+        game = Game(name=name, description=description, type=game_type)
+        db.session.add(game)
+        db.session.commit()
+        return {"message": "Game added successfully", "game": game.name}, 201
+
+    @admin_required
+    def delete(self):
+        """Delete a game"""
+        data = request.get_json()
+        game_id = data.get("id")
+
+        game = Game.query.get(game_id)
+        if not game:
+            return {"error": "Game not found"}, 404
+
```
@prasidha1 
Adding of games will be done by us developers ryt.. admin  cannot add or delete a game.

prasidha1 commented:
closing this branch as merging the same with feature#18_Admin dashboard

prasidha1 commented:
Kindly review and merge the branch

prasidha1 commented:
This is admin page's backend. Kindly review and merge with main

prasidha1 commented:
Feature #22: Admin Dashboard. 
Description
This feature implements the Admin Dashboard, allowing administrators to manage games, view statistics, and control game settings.

Changes Made
Backend:

Added a new route /admin/dashboard in [admin.py] to fetch admin dashboard data, including:
Total users.
Total games.
Recent activities.
Game scores.
Rate limits.
Added a toggle to activate or deactivate games.
Updated blueprint for game-related operations.

PS: Some part of the code has been changed in feature 18 PR
